pilot/pkg/model/service.go

```diff
@@ -45,23 +45,47 @@ type Service struct {
 	// Hostname of the service, e.g. "catalog.mystore.com"
 	Hostname string `json:"hostname"`
 
-	// Address specifies the service IPv4 address of the load balancer
+	// Address specifies the virtual IP address associated with the service.
+	// If empty
 	Address string `json:"address,omitempty"`
 
 	// Ports is the set of network ports where the service is listening for
 	// connections
 	Ports PortList `json:"ports,omitempty"`
 
-	// ExternalName is only set for external services and holds the external
-	// service DNS name.  External services are name-based solution to represent
-	// external service instances as a service inside the cluster.
-	ExternalName string `json:"external"`
-
 	// ServiceAccounts specifies the service accounts that run the service.
 	ServiceAccounts []string `json:"serviceaccounts,omitempty"`
 
-	// LoadBalancingDisabled indicates that no load balancing should be done for this service.
-	LoadBalancingDisabled bool `json:"-"`
+	// MeshExternal (if true) indicates that the service is external to the mesh.
+	// These services are defined using Istio's ExternalService spec.
+	MeshExternal bool
+
+	// Virtual (if true) indicates that the service is not discoverable or directly
+	// routable using its hostname. These are typically used to indicate subsets of
+	// a given service (e.g., v1.reviews.bookinfo.com, where v1 is the name of a subset that
+	// indicates the set of instances associated with version v1 of the reviews service.
+	VirtualService bool
+
+	// Labels is a set of key-value pair metadata associated with each service instance
+	// associated with this service. For virtual services, the set of labels will be a super set
+	// of the set of labels associated with the respective parent service.
+	Labels Labels
```

Comment:
We also have labels as part of `model.ServiceInstance`. What is the difference between the two? If these are a superset of the instance labels, how do we handle conflicts?

Reply:
Note: its a superset of labels associated with the parent service (not instance). 
This is essentially codifying what we currently compute (label selectors). 
Instances can have 100 labels, while the service will have labels that were used to select these instances. 
In platforms like Consul/Eureka, we might probably inject a synthetic label for each instances (e.g., `service: foo.com`) and use this synthetic label as the service label.

If a service is a virtual service (i.e. reviews-v1 for example), the labels here will be a superset of the labels associated with the actual reviews service (e.g., `version: alpha, app: reviews` for the v1.reviews virtual service and `app: reviews` for the actual (non virtual) reviews service. )